### PR TITLE
port dotnet/corefx#10792

### PR DESCRIPTION
**Pull request by anurse**
This just ports the change from dotnet/corefx#10792 to our repo. It also means we can get rid of the warning pragmas we had to add 🎉 .

/cc @stephentoub @Tratcher

### CODE REVIEW

File: samples/Dummy/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Dummy Project
+
+This was the quickest and easiest way to stop KoreBuild from breaking. It doesn't like when there **is** a `samples` directory but **none** of the projects have `project.json` files.
```
Comment (Tratcher) [inline samples/Dummy/README.md]: Wait, why don't we have a Hello World sample anymore?

Comment (Tratcher) [inline samples/Dummy/README.md]: i.e. why not just add a real Hello World sample?

Comment (Tratcher) [inline samples/Dummy/README.md]: https://github.com/aspnet/Entropy/blob/dev/samples/Owin.Nowin.WebSockets/Startup.cs

Comment (anurse) [inline samples/Dummy/README.md]: We never had Hello World, we had a Test Client. I can add some kind of echo server sample.

**Comment by anurse** [comment]
:up: :date: - Added an Echo sample and removed the dummy project :)

File: samples/EchoApp/Project_Readme.html [hunk 2]
```diff
@@ -0,0 +1,187 @@
+﻿<!DOCTYPE html>
```
Comment (Tratcher) [inline samples/EchoApp/Project_Readme.html]: Delete readme

File: samples/EchoApp/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+        {
+            loggerFactory.AddConsole();
+
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
```
Comment (Tratcher) [inline samples/EchoApp/Startup.cs]: Note this will bump the repo layering

File: samples/EchoApp/wwwroot/index.html [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        var connectButton = document.getElementById("connectButton");
+        var stateLabel = document.getElementById("stateLabel");
+        var sendMessage = document.getElementById("sendMessage");
+        var sendButton = document.getElementById("sendButton");
+        var sendForm = document.getElementById("sendForm");
+        var commsLog = document.getElementById("commsLog");
+
+        var socket;
+
+        var scheme = document.location.protocol == "https" ? "wss" : "ws";
```
Comment (campersau) [inline samples/EchoApp/wwwroot/index.html]: `https:`

File: samples/EchoApp/Startup.cs (hunk 3 above)
Comment (Tratcher) [inline samples/EchoApp/Startup.cs]: Nevermind, it's already at a high layer

File: samples/EchoApp/wwwroot/index.html [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        var port = document.location.port != 80 ? (":" + document.location.port) : "";
```
Comment (campersau) [inline samples/EchoApp/wwwroot/index.html]: `document.location.port` is empty if it is the default port so:
```js
var port = document.location.port ? ":" + document.location.port : "";
```

File: samples/EchoApp/project.json [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    "Microsoft.AspNetCore.Diagnostics": "1.1.0-*",
+    "Microsoft.AspNetCore.Server.IISIntegration": "1.1.0-*",
+    "Microsoft.AspNetCore.Server.Kestrel": "1.1.0-*",
+    "Microsoft.Extensions.Logging.Console": "1.1.0-*",
+    "Microsoft.AspNetCore.StaticFiles": "1.1.0-*",
+    "Microsoft.AspNetCore.WebSockets": "0.2.0-*"
+  },
+
+  "tools": {
+    "Microsoft.AspNetCore.Server.IISIntegration.Tools": "1.0.0-preview2-final"
```
Comment (Tratcher) [inline samples/EchoApp/project.json]: *?

**Comment by Tratcher** [comment]
:shipit:

**Pull request merged**